RecoEgamma/EgammaTools/python/egammaGainSwitchFixToolsForPAT_cff.py

```diff
@@ -0,0 +1,157 @@
+import FWCore.ParameterSet.Config as cms 
+
+def replace_input_tag_module_label(tag,names_to_replace,suffix):
+    if type(tag)==str: #having some problems with things appearing as strings....
+        if tag in names_to_replace:
+            tag = tag+suffix
+    else:
+        if tag.getModuleLabel() in names_to_replace:
+            tag.setModuleLabel(tag.getModuleLabel()+suffix)
+      
+#replaces all input tags in the module which match modules to replace with that module name + suffix
+def replace_input_tags(process,modname,pset,modules_to_replace,suffix):
+    for paraname in pset.parameterNames_():
+        para = pset.getParameter(paraname)
+        if para.pythonTypeName()=="cms.PSet":
+            replace_input_tags(process,modname,para,modules_to_replace,suffix)
+        elif para.pythonTypeName()=="cms.VPSet":
+            for newpset in para:
+                replace_input_tags(process,modname,newpset,modules_to_replace,suffix)
+        elif para.pythonTypeName()=="cms.InputTag":
+            replace_input_tag_module_label(para,modules_to_replace,suffix)            
+        elif para.pythonTypeName()=="cms.VInputTag":
+            for tag in para: 
+                replace_input_tag_module_label(tag,modules_to_replace,suffix)
+        
+                                                  
+#clones all these egamma pat modules and then updates the tags of these modules
+#to point to the other cloned modules
+def miniAOD_addOrginalEGamma(process,suffix):
+    modules_to_clone=["slimmedElectrons",
+                      "selectedPatElectrons",
+                      "patElectrons",
+                      "slimmedPhotons",
+                      "selectedPatPhotons",
+                      "patPhotons",
+                      "reducedEgamma",
+                      "egmGsfElectronIDs",
+                      "electronMVAValueMapProducer",
+                      "electronRegressionValueMapProducer",
+                      "egmPhotonIDs",
+                      "photonIDValueMapProducer",
+                      "photonRegressionValueMapProducer",
+                      "photonMVAValueMapProducer",
+                      #"particleBasedIsolation",
+                      #"PhotonIDProdGED",
+                      #"eidLoose",
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant